### ISSUE TITLE
BelongsToMany with pivot simply doesn't work according to documentation

### ISSUE THREAD

**Issue by kirkbushell** [laravel/nova-issues#3462]
- Laravel Version: 8.40.0
- Nova Version: 3.25.0
- PHP Version: 7.4.20
- Database Driver & Version: MySQL
- Operating System and Version: Mac OSX 11.4
- Browser type and version: Chrome 91.0.4472.114

### Description:
Despite recent discussions I could find from February, defining a BelongsToMany field on a nova resource, even when defining that relationship (withPivot fields) on both models, does not appear to respect it. Ie. It never shows the fields in the index view of a parent resource.

The only way I could try and get around this, is to define the fields on the child resource, but this for me poses other problems as it doesn't appear possible to restrict the pivot again based on the parent resource. Ie. It appears to be a separate set of queries.

In short, the documentation states that we can define a BelongsToMany and bring in pivot fields, but this does not appear to be working.

### Detailed steps to reproduce the issue on a fresh Nova installation:

Setup a BelongsToMany relationship, try and get the additional fields to show on a related resource.

**Comment by kirkbushell** [comment]
Closed - discovered a bug with my saves while looking at the wrong record.
